src/nanoarrow/utils_test.cc

```diff
@@ -543,6 +543,36 @@ TEST(DecimalTest, DecimalRoundtripBitshiftTest) {
   ArrowBufferReset(&buffer);
 }
 
+// test case adapted from
+// https://github.com/apache/arrow/blob/main/go/arrow/float16/float16_test.go
+TEST(HalfFloatTest, FloatToHalfFloat) {
+  uint16_t cases_bits[] = {
+    0x3c00,
+    0x4000,
+    0xc000,
+    0x0000,
+    0x5b8f,
+    0xdb8f,
+    0x48c8,
+    0xc8c8,
+  };
+  float cases_float[] = {
```

Comment:
A few more test cases that would be easy to add here:
- `-0` -> `0x8000`
- `+inf` -> `0x7c00`
- `-inf` -> `0xfc00`